### ISSUE TITLE
cross entropy loss function

### ISSUE THREAD

**Issue by Laksh-Mendpara** [madhavkataria1010/ics_majorproject#2]
improve performance by updating the loss function to cross entropy loss. Also change update the eqution to calculate gradient for backpropogation accordingly
